prog/main.go

```diff
@@ -134,6 +138,62 @@ type appFlags struct {
 	consulInf       string
 }
 
+type containerLabelFiltersPositive []app.APITopologyOption
+
+func (filters *containerLabelFiltersPositive) String() string {
+	return fmt.Sprint(*filters)
+}
+
+func (filters *containerLabelFiltersPositive) Set(value string) error {
+	*filters = append(*filters, makeAPITopologyOption(true, value, len(*filters)))
+	return nil
+}
+
+type containerLabelFiltersNegative []app.APITopologyOption
+
+func (filters *containerLabelFiltersNegative) String() string {
+	return fmt.Sprint(*filters)
+}
+
+func (filters *containerLabelFiltersNegative) Set(value string) error {
+	*filters = append(*filters, makeAPITopologyOption(false, value, len(*filters)))
+	return nil
+}
+
+func makeAPITopologyOption(positive bool, value string, filterNumber int) app.APITopologyOption {
+	u := fmt.Sprintf("%q", value)
+	u = u[1 : len(u)-1]
+
+	indices := colonFinder.FindStringIndex(u)
+	if len(indices) == 0 {
+		log.Infof("No unescaped colon found. This is needed to separate the title from the label")
+		return app.APITopologyOption{}
+	}
+	if len(colonFinder.FindAllStringIndex(u, -1)) > 1 {
+		log.Infof("Escape colons that are part of the title and label")
+		return app.APITopologyOption{}
+	}
+
+	title := unescapeColons.ReplaceAllString(u[0:indices[0]+1], `:`)
+	label := unescapeColons.ReplaceAllString(u[indices[1]:], `:`)
```

Comment:
You also need to unescape the backslash, `\` we've discussed this, and I even provided you with a regular expression to do both, sigh ...